ci/buildspec_main.yml

```diff
@@ -0,0 +1,28 @@
+# AWS CodeBuild config
+# This file is referenced in the infra repo as a CodeBuild buildspec target.
+
+# These 3 variables are set in the codebuild project definition's environment secion.
+# ECR_REGION
+# ECR_ACCOUNT_ID
+# ECR_NAMESPACE
+
+version: 0.2
+env:
+  shell: bash
+phases:
+  install:
+    runtime-versions:
+      nodejs: 16
+  pre_build:
+    commands:
+      - aws ecr get-login-password --region "${ECR_REGION}" | docker login --username AWS --password-stdin "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com"
+
+  build:
+    commands:
+      # Backend
+      - docker build backend/core -f build/docker/Dockerfile.backend --target tests
+      - docker build backend/core -f build/docker/Dockerfile.backend --target run
```

Comment:
You'll want to tag the local images too.

`docker build backend/core -f build/docker/Dockerfile.backend --target run -t doorway/backend:run .`

Reply:
The `latest` image is the same thing as a `run` tagged image. Is there a reason we need the run tagged image in the docker cache?

Reply:
For this draft, we're running unit tests(if possible), then building and pushing the `run` image. It sounds like at some point, we'll want to push the `test` images to ECR- is this accurate?

Reply:
That's a possibility.  The main reason why we might want to do that is if we want to run integration test in dev (with dummy data) _and_ staging (with production/production-like data).